tests/test_math_kvecgenerator.cc

```diff
@@ -0,0 +1,280 @@
+        v11m, v110, v111;
+    Eigen::MatrixXd * solutions[3] = {&kvec_sol0, &kvec_sol1, &kvec_sol2};
+
+    // For various cutoff radii close to the thresholds after which
+    // we would get further vectors, we compare the obtained vectors
+    // to the correct solution. The variable i runs over the three intervals
+    double cutofflims[4] = {1., math::SQRT_TWO, math::SQRT_THREE, 2.};
+    double cutoffeps = 1e-12;
+    for (int i = 0; i < 3; i++) {
+      math::Kvectors k_vectorsmin(cutofflims[i] + cutoffeps, reciprocalcell,
+                                  true);
+      BOOST_CHECK_EQUAL(k_vectorsmin.get_kvectors(), *solutions[i]);
+      math::Kvectors k_vectorsmax(cutofflims[i + 1] - cutoffeps, reciprocalcell,
+                                  true);
+      BOOST_CHECK_EQUAL(k_vectorsmax.get_kvectors(), *solutions[i]);
+    }
+
+    // Check whether the code also works if we use a different, highly
+    // distorted basis to describe the same lattice
+    Eigen::Matrix3d cell_distortedbasis;
+    cell_distortedbasis << 1, 0, 0, 5, 1, 0, 5, 5, 1;
+
+    // Define the vectors that should be obtained from this procedure
+    // Note that the obtained vectors should be the same as for the
+    // undistorted lattice, but their order and representant out of
+    // +k and -k will be different
+    Eigen::MatrixXd kvec_sol3(3, 3), kvec_sol4(9, 3);
+    kvec_sol3 << v100, -v010, v001;
+    kvec_sol4 << v100, -v110, -v010, v1m0, v011, -v10m, v001, v101, -v01m;
+    Eigen::MatrixXd * solutions_distorted[2] = {&kvec_sol3, &kvec_sol4};
+
+    // Run the algorithm and compare the obtained k-vectors to the
+    // correct ones
+    for (int i = 0; i < 2; i++) {
+      math::Kvectors k_vectorsmin(cutofflims[i] + cutoffeps,
+                                  cell_distortedbasis, true);
+      BOOST_CHECK_EQUAL(k_vectorsmin.get_kvectors(), *solutions_distorted[i]);
+      math::Kvectors k_vectorsmax(cutofflims[i + 1] - cutoffeps,
+                                  cell_distortedbasis, true);
+      BOOST_CHECK_EQUAL(k_vectorsmax.get_kvectors(), *solutions_distorted[i]);
+    }
+  }  // end test case
+
+  /* ---------------------------------------------------------------------- */
+  /**
+   * @brief Check that all obtained k-vectors lie inside the ball
+   * and that the norms computed by the kvecgenerator are correct.
+   */
+  BOOST_FIXTURE_TEST_CASE(kvecgen_allvecs_in_ball_test, KvecgenRefFixture) {
+    // For each cell: get the required quantities for the tests
+    double cutoff = 6.28;
+    for (auto & cell : this->cells) {
+      math::Kvectors kvectors(cutoff, cell);
+      int nvec = kvectors.get_numvectors();
+      Eigen::MatrixXd kvecs = kvectors.get_kvectors();
+      Eigen::VectorXd kvals = kvectors.get_norms();
+
+      // Make sure that all vector do indeed lie within
+      // the cutoff and that the lengths computed by the kvecgenerator
+      // agree with those values.
```

Comment:
This comment is duplicated from the `@brief` above, no need to have it around twice

Reply:
Still the case 😃!